extensions/amp-story/1.0/test/test-amp-story-quiz.js

```diff
@@ -277,5 +299,63 @@ describes.realWin(
         'i-amphtml-story-quiz-option-selected'
       );
     });
+
+    it('should handle the percentage pipeline', async () => {
+      env.sandbox
+        .stub(requestService, 'executeRequest')
+        .resolves(getMockReactionData());
+
+      ampStoryQuiz.element.setAttribute('endpoint', 'http://localhost:8000');
+
+      populateStandardQuizContent(win, ampStoryQuiz.element);
+      ampStoryQuiz.buildCallback();
+      await ampStoryQuiz.layoutCallback();
+
+      const quizElement = ampStoryQuiz.getQuizElement();
+      const quizOptions = quizElement.querySelectorAll(
+        '.i-amphtml-story-quiz-option'
+      );
+
+      const percentageOption0 = quizOptions[0].querySelector(
+        '.i-amphtml-story-quiz-percentage-text'
+      );
+
+      expect(percentageOption0.textContent).to.equal('30%');
+
+      const percentageOption3 = quizOptions[3].querySelector(
+        '.i-amphtml-story-quiz-percentage-text'
+      );
+
+      expect(percentageOption3.textContent).to.equal('10%');
+    });
+
+    it('should preprocess percentages properly', () => {
+      const responseData1 = getMockReactionData()['data'];
+
+      const percentages1 = ampStoryQuiz.preprocessPercentages_(responseData1);
+
+      expect(percentages1).to.deep.equal([30, 30, 30, 10]);
+    });
+
+    it('should preprocess percentages preserving ties', () => {
+      const responseData2 = generateResponseDataFor([3, 3, 3]);
+      const percentages2 = ampStoryQuiz.preprocessPercentages_(responseData2);
+
+      expect(percentages2).to.deep.equal([33, 33, 33]);
+    });
+
+    it('should preprocess percentages preserving order', () => {
+      const responseData3 = generateResponseDataFor([255, 255, 245, 245]);
+      const percentages3 = ampStoryQuiz.preprocessPercentages_(responseData3);
+
+      expect(percentages3).to.deep.equal([26, 26, 24, 24]);
+    });
+
+    it('should preprocess percentages handling rounding edge cases', () => {
+      const responseData4 = generateResponseDataFor([335, 335, 330]);
```

Comment:
I'm curious. What should happen if we called `generateResponseDataFor([335, 335, 329]);

Would it create an artificial tie by rounding up the 329/1000 to also be `33%`.

Reply:
So in this case it would actually end up as `33% 33% 32%`, because it would be out of 999, but if we had a 4th option with 1 vote it would artificially create a tie.

This is because the heuristic I was using (a total higher than `100` includes some values with remainder `0.50`) doesn't hold exactly. I'm gonna modify the portion of the algorithm that uses that assumption, nice catch.

Reply:
I adjusted this so in the above 100 case it scales down the remainders but keeps their sizes relative to each other (I divided by three because that's the maximum amount over the remainders can add up to).

As far as creating ties where there aren't any, I think it's acceptable to do if values are close, though I am considering possibilities for ways to get something like `[335, 335, 329, 1]` to round to `[34%, 34%, 32%, 0%]`